x/gamm/twap/types/utils.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	fmt "fmt"
+	"sort"
+	time "time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v10/osmoutils"
+)
+
+func NewTwapRecord(k AmmInterface, ctx sdk.Context, poolId uint64, denom0 string, denom1 string) TwapRecord {
+	if !(denom0 > denom1) {
+		panic(fmt.Sprintf("precondition denom0 > denom1 not satisfied. denom0 %s | denom1 %s", denom0, denom1))
+	}
```

Comment:
```suggestion
	if !(denom0 > denom1) {
		panic(fmt.Sprintf("precondition denom0 > denom1 not satisfied. denom0 %s | denom1 %s", denom0, denom1))
	}
		denoms, err := k.GetPoolDenoms(ctx, poolId)
	if err != nil {
		panic(err)
	}
	for _, denom := range denoms {
		if denom != denom0 || denom != denom1 {
			panic(fmt.Sprintf("provided poolId is not composed of the provided denoms"))
		}
	}
```

Reply:
Unsure of what the caller of this function looks like, but we can ensure the provided poolId is composed of the provided denoms

Reply:
the AMM keeper will handle erroring in this case for us, in the subsequent spot price lines